openmc/_xml.py

```diff
@@ -1,22 +1,38 @@
-def clean_indentation(element, level=0, spaces_per_level=2):
-    """
-    copy and paste from https://effbot.org/zone/element-lib.htm#prettyprint
-    it basically walks your tree and adds spaces and newlines so the tree is
-    printed in a nice way
+def clean_indentation(element, level=0, spaces_per_level=2, trailing_indent=True):
+    """Set indentation of XML element and its sub-elements.
+    Copied and pasted from https://effbot.org/zone/element-lib.htm#prettyprint.
+    It walks your tree and adds spaces and newlines so the tree is
+    printed in a nice way.
+
+    Parameters
+    ----------
+    level : int
+        Indentation level for the element passed in (default 0)
+    spaces_per_level : int
+        Number of spaces per indentation level (default 2)
+    trailing_indent : bool
+        Whether or not to add indentation after closing the element
+
     """
     i = "\n" + level*spaces_per_level*" "
 
+    # ensure there's awlays some tail for the element passed in
+    if not element.tail:
+        element.tail = ""
+
     if len(element):
         if not element.text or not element.text.strip():
             element.text = i + spaces_per_level*" "
-        if not element.tail or not element.tail.strip():
+        if trailing_indent and (not element.tail or not element.tail.strip()):
             element.tail = i
         for sub_element in element:
+            # `trailing_indent` intentionally not passed to the recursive call.
+            # it only applies to the element for the initial of this function.
```

Comment:
It's just the opposite. We don't want to forward a value of `False` here because any recursive call is for a child element, in which case we know there _should_ be trailing indentation so that its parent element has the correct indentation. 

I'll improve that comment's clarity. 

Maybe there's 
a better pattern I should be using? This was the simplest and easiest thing that came to mind.

Reply:
Gotcha. The pattern is fine, but the comment could indeed be improved